scope.go

```diff
@@ -1230,6 +1227,40 @@ func (scope *Scope) getColumnAsArray(columns []string, values ...interface{}) (r
 	return
 }
 
+func (scope *Scope) getColumnAsArrayUnique(columns []string, values ...interface{}) (results [][]interface{}) {
+	unfilteredResults := scope.getColumnAsArray(columns, values...)
+
+	rootMap := map[interface{}]interface{}{}
```

Comment:
If I understand you correctly, the main point is in `O(log(N))` lookups in case of multiple primary keys. So its basically a map of map of empty structs (which occupy zero memory on contrary to bools)  when primary key consists of two columns. Keys of first map are unique values for first column and keys of second-level maps are all unique values for second column which exist together with first column value, and so on.